Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -187,7 +187,9 @@ bool WidthIterator::hasExtraSpacing() const
 
 static void resetGlyphBuffer(GlyphBuffer& glyphBuffer, GlyphBufferStringOffset index)
 {
+#if USE(CG)
     ASSERT(index >= 0);
```

Comment:
Would this work?
```
ASSERT(!std::is_signed_v<GlyphBufferStringOffset> || index >= 0);
```

Reply:
Doesn't work here, probably because it's just static analysis. Regardless, I'm a bit worried about having a data type that has a different signedness depending on the architecture. Not sure if it's a good idea, other bounding bugs might go unnoticed. Any reason why this needs to be signed in Apple ports?